pkg/linter/linter.go

```diff
@@ -156,6 +161,40 @@ func worldWriteableLinter(_ LinterContext, path string, d fs.DirEntry) error {
 	return nil
 }
 
+func strippedLinter(lctx LinterContext, path string, d fs.DirEntry) error {
+	if !d.Type().IsRegular() {
+		// Don't worry about non-files
+		return nil
+	}
+
+	info, err := d.Info()
+	if err != nil {
+		return err
+	}
+
+	ext := filepath.Ext(path)
+	mode := info.Mode()
+	if mode&0111 == 0 && !isObjectFileRegex.MatchString(ext) {
+		// Not an executable or library
+		return nil
+	}
+
+	file, err := elf.Open(filepath.Join(lctx.realfspath, path))
```

Comment:
Can we use `elf.NewFile` and extract the image from the FS, instead of having this take `realfspath`? This might make it easier to run this on arbitrary FSes, e.g., non-unpacked tars from apks streaming in over HTTP.

Reply:
Yeah, sure. I'll modify it to take the fs path.

Reply:
So there's a problem with this approach:

fs.Open doesn't implement io.ReaderAt. >_<

Reply:
Fixed it with a crappy workaround for now.